src/Sarif.Multitool.Library/Rules/SARIF2007.ExpressPathsRelativeToRepoRoot.cs

```diff
@@ -58,7 +58,10 @@ protected override void Analyze(Run run, string runPointer)
                     }
                     else
                     {
-                        this.uriBaseIds.Add(run.VersionControlProvenance[i].MappedTo.UriBaseId);
+                        string validUriBaseId = run.VersionControlProvenance[i].MappedTo.UriBaseId;
+                        this.uriBaseIds.Add(validUriBaseId);
+                        run.OriginalUriBaseIds?.Where(baseId => baseId.Value?.UriBaseId == validUriBaseId).ToList()
```

Comment:
>Where [](http://example.com/codeflow?start=48&length=5)

In general, please don't implement linq queries of this kind in our code base. Instead author patterns that iterate over the original collection a single time.